packages/mobile/src/send/SendAmount.tsx

```diff
@@ -1,254 +1,217 @@
-import { getLocalCurrencyCode, getLocalCurrencyExchangeRate } from 'src/localCurrency/selectors'
-import { HeaderTitleWithBalance, headerWithBackButton } from 'src/navigator/Headers'
+import {
+  getLocalCurrencyCode,
+  getLocalCurrencyExchangeRate,
+  getLocalCurrencySymbol,
+} from 'src/localCurrency/selectors'
 import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { StackParamList } from 'src/navigator/types'
 import { getRecipientVerificationStatus, Recipient } from 'src/recipients/recipient'
-import { RootState } from 'src/redux/reducers'
-import { PaymentInfo } from 'src/send/reducers'
+import useSelector from 'src/redux/useSelector'
 import { getRecentPayments } from 'src/send/selectors'
-import { dailyAmountRemaining, getFeeType } from 'src/send/utils'
+import { dailyAmountRemaining, getFeeType, isPaymentLimitReached } from 'src/send/utils'
 import DisconnectBanner from 'src/shared/DisconnectBanner'
 import { fetchDollarBalance } from 'src/stableToken/actions'
-import { withDecimalSeparator } from 'src/utils/withDecimalSeparator'
+import { stableTokenBalanceSelector } from 'src/stableToken/reducer'
 
 export interface TransactionDataInput {
   recipient: Recipient
   amount: BigNumber
-  reason: string
   type: TokenTransactionType
+  reason: string
   firebasePendingRequestUid?: string | null
 }
 
-const AmountInput = withDecimalSeparator(
-  withTextInputLabeling<ValidatedTextInputProps<DecimalValidatorProps>>(ValidatedTextInput)
-)
-const CommentInput = withTextInputLabeling<TextInputProps>(TextInput)
+type RouteProps = StackScreenProps<StackParamList, Screens.SendAmount>
+type Props = RouteProps
 
-interface State {
-  amount: string
-  reason: string
-}
+const { decimalSeparator } = getNumberFormatSettings()
 
-type OwnProps = StackScreenProps<StackParamList, Screens.SendAmount>
-type Props = StateProps & DispatchProps & OwnProps & WithTranslation
+function SendAmount(props: Props) {
+  const dispatch = useDispatch()
 
-interface StateProps {
-  dollarBalance: string
-  estimateFeeDollars: BigNumber | undefined
-  defaultCountryCode: string
-  feeType: FeeType | null
-  localCurrencyCode: LocalCurrencyCode
-  localCurrencyExchangeRate: string | null | undefined
-  recipient: Recipient
-  recipientVerificationStatus: RecipientVerificationStatus
-  addressValidationType: AddressValidationType
-  recentPayments: PaymentInfo[]
-}
+  const isRequest = props.route.params?.isRequest ?? false
+  const recipient = props.route.params?.recipient
 
-interface DispatchProps {
-  fetchDollarBalance: typeof fetchDollarBalance
-  showMessage: typeof showMessage
-  showError: typeof showError
-  hideAlert: typeof hideAlert
-  fetchAddressesAndValidate: typeof fetchAddressesAndValidate
-}
+  React.useEffect(() => {
+    dispatch(fetchDollarBalance())
+    if (recipient.e164PhoneNumber) {
+      dispatch(fetchAddressesAndValidate(recipient.e164PhoneNumber))
```

Comment:
This leads to "Enter PIN" screen appearing right when user is shown "Enter Amount" screen. This is not the best user experience, but I do not know how to solve it better. @jmrossy is that something related to Phone Privacy, right?